os/using-environment-variables-in-systemd-units.md

```diff
@@ -0,0 +1,116 @@
+
+You can solve this problem without new cluster bootstrapping. Run `etcdctl member add node4 http://10.0.1.13:2380` and remember its output (we will use it later):
+
+```
+added member 9bf1b35fc7761a23 to cluster
+
+ETCD_NAME="node4"
+ETCD_INITIAL_CLUSTER="node1=http://10.0.1.10:2380,node2=http://10.0.1.11:2380,node3=http://10.0.1.12:2380,node4=http://10.0.1.13:2380"
+ETCD_INITIAL_CLUSTER_STATE=existing
+```
+
+Already defined in `20-cloudinit.conf` `ETCD_DISCOVERY` conflicts with `ETCD_INITIAL_CLUSTER` environment variable, so we have to clean it. We can do that overriding `20-cloudinit.conf` by `99-restore.conf` drop-in with the `Environment="ETCD_DISCOVERY="` string.
+
+The complete example will look this way. On `node4` CoreOS host create temporarily systemd drop-in unit `/run/systemd/system/etcd2.service.d/99-restore.conf` with the content below (we use variables from `etcdctl member add` output):
+
+```
+[Service]
+# remove previously created proxy directory
+ExecStartPre=/usr/bin/rm -rf /var/lib/etcd2/proxy
+# here we clean previously defined ETCD_DISCOVERY environment variable, we don't need it as we've already bootstrapped etcd cluster and ETCD_DISCOVERY conflicts with ETCD_INITIAL_CLUSTER environment variable
+Environment="ETCD_DISCOVERY="
+Environment="ETCD_NAME=node4"
+Environment="ETCD_INITIAL_CLUSTER=node1=http://10.0.1.10:2380,node2=http://10.0.1.11:2380,node3=http://10.0.1.12:2380,node4=http://10.0.1.13:2380"
+Environment="ETCD_INITIAL_CLUSTER_STATE=existing"
+```
+
+Run `sudo systemctl daemon-reload` and `sudo systemctl restart etcd2` to apply your changes. You will see that your proxy node became cluster member:
+
+```
+etcdserver: start member 9bf1b35fc7761a23 in cluster 36cce781cb4f1292
+```
+
+Once your proxy node became member node and `etcdctl cluster-health` shows healthy cluster, you can remove your temporarily drop-in `sudo rm /run/systemd/system/etcd2.service.d/99-restore.conf && sudo systemctl daemon-reload`.
+
+## EnvironmentFile directive
+
+EnvironmentFile similar to Environment directive but reads the environment variables from a text file. The text file should contain new-line-separated variable assignments. Empty lines and lines starting with ; or # will be ignored, which may be used for commenting. A line ending with a backslash will be concatenated with the following one, allowing multiline variable definitions. The parser strips leading and trailing whitespace from the values of assignments, unless you use double quotes ("). 
+
+The argument passed should be an absolute filename or wildcard expression, optionally prefixed with "-", which indicates that if the file does not exist, it will not be read and no error or warning message is logged. This option may be specified more than once in which case all specified files are read. If the empty string is assigned to this option, the list of file to read is reset, all prior assignments have no effect.
```

Comment:
A lot of this seems very specific. I don't think it should be in the second paragraph when we have much more useful info we could surface.

Should we move this towards the bottom or just link to upstream docs for this level of detail?